oauth2client/contrib/metadata.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Thin wrapper class for talking to the GCE Metadata Server."""
+import datetime
+import httplib2
+import json
+
+from six.moves import http_client
+
+from oauth2client._helpers import _from_bytes
+from oauth2client.client import _UTCNOW
+from oauth2client.client import HttpAccessTokenRefreshError
+
+METADATA_ROOT = 'http://metadata.google.internal/computeMetadata/v1/'
+METADATA_HEADERS = {'Metadata-Flavor': 'Google'}
+
+
+def get(path, recursive=True, http_request=None, root=METADATA_ROOT):
+    if path is None:
+        path = []
+
+    if not http_request:
+        http_request = httplib2.Http().request
+
+    r_string = '/?recursive=true' if recursive else ''
+    full_path = root + '/'.join(path) + r_string
```

Comment:
That's fine, you can skip urljoin, but is there really any reason for path to be a list rather than string? Seems to violate the principle of least surprise.